src/main/java/com/pkb/unit/Unit.java

```diff
@@ -398,26 +530,30 @@ protected void failed() {
     }
 
 
+    /**
+     * Creates a Message with the previous and current {@link State}, and with the previous and
+     * the current {@link DesiredState} and the given comment.
+     *
+     * @param previous the {@link State} of the Unit before change. Pass the current state
+     *                 if not changed.
+     * @param previousDesired the {@link DesiredState} of the Unit before change. Pass the
+     *                        current state if not changed.
+     * @param comment a text to include into the message. Pass the current state if not
```

Comment:
I don't understand the second part: "Pass the current state if not changed" - this is a comment and not a State. What do you mean here?

Reply:
This means if there is no state change then the current and previous states are the same.
Great you noticed this. Actually if no state change happened then users should call the same method without parameters to advertise the current states so this part you don't understand can be omitted.